### PR TITLE
rename selem to footprint everywhere

### PR DESCRIPTION
**Pull request by grlee77**
## Description

In the discussion in #4154 it was decided that `footprint` should be less ambiguous than `selem` for use in all rank filters and morphology functions.

This PR makes that change

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

File: skimage/morphology/selem.py
Comment (jni) [inline skimage/morphology/selem.py:13]: Lovely.

I wonder though whether we should not instead encourage directly using `from skimage.morphology import footprints; footprint = footprints.square(5)`. To me this addresses the point in #5439 about grouping functions according to "purpose and equivalence classes". `morphology.square()` is also less descriptive than `footprints.square`.

Perhaps something to add to a dedicated issue for discussion (tracked from #5439) rather than hold up this PR.

Comment (jni) [reply skimage/morphology/selem.py:13]: (But, a quick fix here would be to remain agnostic in the prescription: instead of "should be imported from skimage.morphology instead", we could write "skimage.morphology.selem has been moved to skimage.morphology.footprints.") (And yes I have a mild preference for "footprints" over "footprint", because the latter is easily confused with a variable/kwarg name.)

File: skimage/morphology/tests/test_gray.py
Comment (jni) [inline skimage/morphology/tests/test_gray.py:286]: Gotta love this function 😂

**Review by jni — APPROVED** [review]
Awesome. 🙏 I've left one minor comment about the future of the API, but happy to merge this as-is and leave that discussion for later.

**Comment by jni** [comment]
Note: if there are no further updates I'll merge this by the end of the week.

File: skimage/measure/_label.py
Comment (grlee77) [reply skimage/measure/_label.py:25]: No, but all instances of structure as a `kwarg` seem to be when calling `scipy.ndimage` functions. I doubt there will be much enthusiasm for renaming those upstream (at least there was resistance to renaming the first argument from `input` to `image` in the past).

File: skimage/morphology/selem.py
Comment (grlee77) [reply skimage/morphology/selem.py:13]: I was not sure if this `selem`/`footprint` module within `filters`, `morphology`, etc. was really intended to be public or not. Specifically, these functions are [grouped in with all other morphology functions in the docs](https://scikit-image.org/docs/stable/api/skimage.morphology.html).

I am fine with renaming to `footprints.py` and importing from there, but wonder if we can perhaps follow up later to make these logical groupings (`binary` vs. `gray` vs. `footprints`) clearer in the docs.

**Comment by grlee77** [comment]
Thanks for reviewing @jni. I made the suggested rename and warning changes. As far as I could tell, the uses of `structure` arise of `scipy.ndimage` functions, not ours.

Two other things were also updated since you reviewed:
1.) renamed test_selem.py -> test_footprints.py
2.) There were still tons of mentions of "structuring element" in the docstrings! These now refer to footprint

File: skimage/measure/_label.py
Comment (jni) [reply skimage/measure/_label.py:25]: Oh my apologies, I thought this was `measure.label`. 🤦

File: skimage/morphology/selem.py
Comment (jni) [reply skimage/morphology/selem.py:12]: Yes, following things up later is totally fine, but that's why I'm saying, let's not be prescriptive in the deprecation message, because we might change our mind during those discussions.

**Comment by jni** [comment]
Great. Totally happy with the changes now. I'll merge this in <24h if no one has stepped in to review+merge before then.

**Review by rfezzani — APPROVED** [review]
Thank you @grlee77, merging :wink:

**Pull request merged**
